homeassistant/components/flux_led/__init__.py

```diff
@@ -1 +1,95 @@
-"""The flux_led component."""
+"""The Flux LED/MagicLight integration."""
+import copy
+from datetime import timedelta
+import logging
+
+from flux_led import BulbScanner
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICES, CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.util import Throttle
+
+from .const import CONF_AUTOMATIC_ADD, DEFAULT_NETWORK_SCAN_INTERVAL, SIGNAL_ADD_DEVICE
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["light"]
+
+
```

Comment:
It looks like the id is the mac address 
```
 % python3 -m flux_led -sSti                 
  warn(RuntimeWarning(msg))
18B905C5239B [192.168.214.166] ON  [Color: (0, 255, 0) White: 0 raw state: 129,51,35,97,35,1,0,255,0,0,9,0,240,84,]
  Timer #1: Unset
  Timer #2: Unset
  Timer #3: Unset
  Timer #4: Unset
  Timer #5: Unset
  Timer #6: Unset

```

Reply:
We could do something similar to `tplink`

https://github.com/home-assistant/core/blob/dev/homeassistant/components/tplink/__init__.py#L107

Scan for devices in `async_setup` and create discovery flows for each one
https://github.com/home-assistant/core/blob/718f8d8bf796e9bb7cbdc29a7f2d19d79e9f5927/homeassistant/components/tplink/__init__.py#L65

If the ip address changes we update the config entry.

This would also allow setting a unique id on the config entry which could be used for the device that would meet the unique id requirements

Reply:
That's correct it is the MAC.